src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -126,7 +151,16 @@ public function select(Composer $composer, $verbose)
 
         $repos = $composer->getRepositoryManager()->getRepositories();
         $pool = new Pool($this->minimumStability);
-        foreach ($repos as $repo) {
+        foreach ($repos as $key => $repo) {
+            if ($this->hasRepositoryFilter()) {
+                $repoConfig = $repo->getRepoConfig();
```

Comment:
It's not safe to use this method. It's not defined in the RepositoryInterface.
E.g.:
I ran into the problem that the Composer\Repository\PearRepository lacks this method.